csharp/private/macros/setup_basic_nuget_package.bzl

```diff
@@ -1,22 +1,34 @@
 load("//csharp/private:providers.bzl", "CSharpAssembly")
 load("//csharp/private:rules/imports.bzl", "import_library", "import_multiframework_library")
 
+NugetOnlyFrameworks = {
+    "net11": "net20",
+    "net30": "net40",
+    "net35": "net40",
+    "net403": "net45",
+}
+
 def _import_dll(dll, has_pdb, imports):
     path = dll.split("/")
 
     tfm = path[1]
-
-    # Ignore frameworks we don't support (like net35)
-    if tfm not in CSharpAssembly:
-        return
-
+    nuget_only_framework = False
     lib_name = path[-1].rsplit(".", 1)[0]
-
     target_name = "%s-%s" % (lib_name, tfm)
 
+    if tfm in NugetOnlyFrameworks:
+        # Some frameworks we support from NuGet packages only, but we do it by
+        # pretending they are the next closest framework.
+        tfm = NugetOnlyFrameworks[tfm]
+        nuget_only_framework = True
+    elif tfm not in CSharpAssembly:
+        # Ignore other frameworks
+        return
+
     if lib_name not in imports:
         imports[lib_name] = {tfm: target_name}
-    else:
+    elif tfm not in imports[lib_name] or not nuget_only_framework:
+        # We don't overwrite for nuget-only frameworks
```

Comment:
Somewhat of a nit: We should theoretically prefer the `net35` over the `net30`. Not sure if you recognize that isn't happening, but decided this was close enough, or if you just didn't notice.

I think the behaviour as is is fine though, and an issue can be spun off.

Reply:
Probably fixable by iterating dlls in reverse order?

Reply:
Maybe. Let's talk about this tomorrow :)